lib/Sema/TypeCheckStorage.cpp

```diff
@@ -1147,8 +1147,9 @@ synthesizeTrivialGetterBody(AccessorDecl *getter, TargetImpl target,
     body.push_back(returnStmt);
   }
 
+  // Don't mark local accessors as type-checked - captures still need to be computed.
```

Comment:
This seems a bit fragile because you're assuming that type checking is idempotent on already-checked AST. Would it be better to insert an explicit call to computeCaptures() inside TypeCheckDeclPrimary.cpp?

Reply:
(And yeah, computeCaptures() should be a request, since there are a few places we have to trigger it manually for synthesized code...)

Reply:
I tried to change this, but I discovered that "re" type checking the accessor is doing more necessary things than just computing captures.

For example, if the `wrappedValue` getter of a local property wrapper is mutating, `buildStorageReference` will generate something like for the local getter:

```
(member_ref_expr implicit type='String' decl=test.(file).Lazy.wrappedValue@test.swift:10:7 [with (substitution_map generic_signature=<Value> (substitution Value -> String))]
  (declref_expr implicit type='Lazy<String>' decl=test.(file).testLocalLazy()._value@test.swift:28:13 direct_to_storage function_ref=unapplied))
```

But it really needs to look like this:

```
(load_expr implicit type='String'
  (member_ref_expr implicit type='@lvalue String' decl=test.(file).Lazy.wrappedValue@test.swift:10:7 [with (substitution_map generic_signature=<Value> (substitution Value -> String))]
    (inout_expr implicit type='inout Lazy<String>'
      (declref_expr implicit type='@lvalue Lazy<String>' decl=test.(file).testLocalLazy()._value@test.swift:28:13 direct_to_storage function_ref=unapplied))))
```

CSApply knows how to do this, but `buildStorageReference` doesn't. Is there a reason why `buildStorageReference` is attempting to build a type-checked expression rather than relying on the constraint system to fill in types?

Reply:
Addressing this in a follow-up PR is fine. I suggest investigating why the mutating getter case works for property wrappers on members of types but not with local variables. It might be an easy fix.